lib/SILGen/SILGenApply.cpp

```diff
@@ -1232,6 +1232,13 @@ class SILGenApply : public Lowering::ExprVisitor<SILGenApply> {
   }
 
   void visitFunctionConversionExpr(FunctionConversionExpr *e) {
+    // If this is a flattening function conversion, emit the expression
+    // directly.
+    if (e->isFlattening()) {
```

Comment:
Imagine you end up with both a flattening and a conversion here -- either you should handle both cases or make it a separate Expr